packages/components/date-picker/src/date-picker-com/basic-year-table.vue

```diff
@@ -59,54 +67,165 @@ const startYear = computed(() => {
+      } else {
+        cell.start = !!(props.minDate && calTime.isSame(props.minDate, 'year'))
+        cell.end = !!(calEndDate && calTime.isSame(calEndDate, 'year'))
+      }
+
+      const isToday = now.isSame(calTime)
+      if (isToday) {
+        cell.type = 'today'
+      }
+      cell.text = index
+      const cellDate = calTime.toDate()
+      cell.disabled =
+        (props.disabledDate && props.disabledDate(cellDate)) || false
+      row[j] = cell
+    }
+  }
+
+  return rows
+})
+
 const focus = () => {
   currentCellRef.value?.focus()
 }
 
-const getCellKls = (year: number) => {
-  const kls: Record<string, boolean> = {}
+const getCellKls = (cell: YearCell) => {
+  const style: Record<string, boolean> = {}
   const today = dayjs().locale(lang.value)
+  const year = cell.text
 
-  kls.disabled = props.disabledDate
+  style.disabled = props.disabledDate
     ? datesInYear(year, lang.value).every(props.disabledDate)
     : false
 
-  kls.current =
-    castArray(props.parsedValue).findIndex((d) => d!.year() === year) >= 0
+  style.today = today.year() === year
 
-  kls.today = today.year() === year
+  if (cell.inRange) {
+    style['in-range'] = true
 
-  return kls
+    if (cell.start) {
+      style['start-date'] = true
+    }
+
+    if (cell.end) {
+      style['end-date'] = true
+    }
+  }
+  return style
 }
 
-const isSelectedCell = (year: number) => {
-  return (
-    (year === startYear.value &&
-      props.date.year() < startYear.value &&
-      props.date.year() > startYear.value + 9) ||
-    castArray(props.date).findIndex((date) => date.year() === year) >= 0 ||
-    castArray(props.parsedValue).findIndex((date) => date?.year() === year) >= 0
-  )
+const isSelectedCell = (cell: YearCell) => {
+  const year = cell.text
+  return castArray(props.date).findIndex((date) => date.year() === year) >= 0
 }
 
 const handleYearTableClick = (event: MouseEvent | KeyboardEvent) => {
-  const clickTarget = event.target as HTMLDivElement
-  const target = clickTarget.closest('td')
-  if (target && target.textContent) {
-    if (hasClass(target, 'disabled')) return
-    const year = target.textContent || target.innerText
-    if (props.selectionMode === 'years') {
-      if (event.type === 'keydown') {
-        emit('pick', castArray(props.parsedValue), false)
-        return
-      }
-      const newValue = hasClass(target, 'current')
-        ? castArray(props.parsedValue).filter((d) => d?.year() !== Number(year))
-        : castArray(props.parsedValue).concat([dayjs(year)])
-      emit('pick', newValue)
+  const target = (event.target as HTMLElement)?.closest(
+    'td'
+  ) as HTMLTableCellElement
+  if (hasClass(target, 'disabled')) return
+
+  const column = target.cellIndex
+  const row = (target.parentNode as HTMLTableRowElement).rowIndex
+
+  const selectedYear = row * 4 + column + startYear.value
+  const newDate = dayjs().year(selectedYear)
+
+  if (props.selectionMode === 'range') {
```

Comment:
The code related to `if (props.selectionMode === 'years') {` should not be deleted here